src/schema.ts

```diff
@@ -163,17 +180,99 @@ export class Schema {
 }
 
 /**
- * @return a summary with the correct distinct property given a max number of bins
+ * @return a summary of the binning scheme determined from the given max number of bins
  */
-function binSummary(maxbins: number, summary: Summary) {
-  const bin = dlBin({
+function binSummary(maxbins: number, summary: Summary, type: Type) {
+  const bin = dl.bins({
     min: summary.min,
     max: summary.max,
     maxbins: maxbins
   });
-  const binSum = extend({}, summary);
-  binSum.distinct = (bin.stop - bin.start) / bin.step;
-  return binSum;
+
+  return binStats(bin, summary, type);
+}
+
+function timeUnitSummary(unit: TimeUnit, summary: Summary, type: Type) {
+  var bin;
+  switch (unit) {
+    // these are the units that dl.bins.date supports
+    case TimeUnit.YEAR:
+    case TimeUnit.MONTH:
+    case TimeUnit.DAY:
+    case TimeUnit.HOURS:
+    case TimeUnit.MINUTES:
+    case TimeUnit.SECONDS:
+      bin = dl.bins.date({
+        min: summary.min,
+        max: summary.max,
+        unit: unit
+      });
+    // let dl.bins.date infer a unit
+    default:
+      bin = dl.bins.date({
+        min: summary.min,
+        max: summary.max
+      });
+  }
+
+  return binStats(bin, summary, type);
+}
+
+/**
+ * @return a new summary based on a new binning scheme and old summary statistics
+ */
+function binStats(bin, summary: Summary, type: Type): Summary {
```

Comment:
Please add appropriate interface for `bin` ... right now it's quite hard to read.